tests/chat_completions_detection.rs

```diff
@@ -0,0 +1,950 @@
+    let detector_name = DETECTOR_NAME_ANGLE_BRACKETS_WHOLE_DOC;
+
+    let messages = vec![
+        Message {
+            content: Some(Content::Text("Hi there!".to_string())),
+            role: Role::User,
+            ..Default::default()
+        },
+        Message {
+            content: Some(Content::Text("Hello!".to_string())),
+            role: Role::Assistant,
+            ..Default::default()
+        },
+    ];
+
+    // Add mocksets
+    let mut detector_mocks = MockSet::new();
+    let mut chat_mocks = MockSet::new();
+
+    let expected_choices = vec![
+        ChatCompletionChoice {
+            message: ChatCompletionMessage {
+                role: messages[0].role.clone(),
+                content: Some("Hi there!".to_string()),
+                refusal: None,
+                tool_calls: vec![],
+            },
+            index: 0,
+            logprobs: None,
+            finish_reason: "NOT_FINISHED".to_string(),
+        },
+        ChatCompletionChoice {
+            message: ChatCompletionMessage {
+                role: messages[1].role.clone(),
+                content: Some("Hello!".to_string()),
+                refusal: None,
+                tool_calls: vec![],
+            },
+            index: 1,
+            logprobs: None,
+            finish_reason: "EOS_TOKEN".to_string(),
+        },
+    ];
+
+    let expected_detections = Some(ChatDetections {
+        input: vec![],
+        output: vec![],
+    });
+
+    let chat_completions_response = ChatCompletion {
+        model: MODEL_ID.into(),
+        choices: expected_choices.clone(),
+        detections: Some(ChatDetections {
+            input: vec![],
+            output: vec![],
+        }),
+        warnings: vec![],
+        ..Default::default()
+    };
+
+    // Add detector input mock
+    detector_mocks.mock(|when, then| {
+        when.post()
+            .path(TEXT_CONTENTS_DETECTOR_ENDPOINT)
+            .json(ContentAnalysisRequest {
+                contents: vec!["Hi there!".into()],
+                detector_params: DetectorParams::new(),
+            });
+        then.json(vec![Vec::<ContentAnalysisResponse>::new()]);
+    });
+    // Add detector output mock
+    detector_mocks.mock(|when, then| {
+        when.post()
+            .path(TEXT_CONTENTS_DETECTOR_ENDPOINT)
+            .json(ContentAnalysisRequest {
+                contents: vec!["Hello!".into()],
+                detector_params: DetectorParams::new(),
+            });
+        then.json(vec![Vec::<ContentAnalysisResponse>::new()]);
+    });
+
+    // Add chat completions mock
+    chat_mocks.mock(|when, then| {
+        when.post()
+            .path(CHAT_COMPLETIONS_ENDPOINT)
+            .json(ChatCompletionsRequest {
+                messages: messages.clone(),
+                model: MODEL_ID.into(),
+                stream: false,
+                ..Default::default()
+            });
+        then.json(chat_completions_response.clone());
```

Comment:
This mocktail method also accepts references so you can avoid the unnecessary clone.
```suggestion
        then.json(&chat_completions_response);
```